WordPress/src/main/java/org/wordpress/android/push/GCMMessageService.java

```diff
@@ -359,34 +359,13 @@ private void buildAndShowNotificationFromNoteData(Context context, Bundle data)
             if (!buildNoteObjectFromPNPayloadAndSaveIt(data)) {
                 // PN payload doesn't have the note or there was an error.
                 // Retrieve the Note obj by calling the REST API
-
-                WordPress.getRestClientUtilsV1_1().getNotification(
-                        wpcomNoteID,
+                NotificationsActions.downloadNoteAndUpdateDB(wpcomNoteID,
```

Comment:
It doesn't look like notes are getting replaced properly in the db when you moderate.

To reproduce:
1. Send a comment notification with more text than the payload can handle.
2. Moderate the comment (Approve or Unapprove).
3. Observe that there is a duplicate in the list.

![device-2016-11-07-092743](https://cloud.githubusercontent.com/assets/789137/20068286/fc00a25c-a4cc-11e6-8ff0-b970b4a67b33.png)